packages/hash/frontend/src/components/layout/PageSidebar/WorkspaceSwitcher.tsx

```diff
@@ -18,38 +33,68 @@ const truncateText = (text: string) => {
 };
 
 export const WorkspaceSwitcher: VFC<WorkspaceSwitcherProps> = () => {
-  const [open, setOpen] = useState(false);
   const buttonRef = useRef<HTMLButtonElement>(null);
+  const popupState = usePopupState({
+    variant: "popover",
+    popupId: "workspace-switcher-menu",
+  });
   const { user } = useUser();
   const { logout } = useLogout();
+  const { query } = useRouter();
+
+  const activeWorkspace = useMemo(() => {
+    const activeAccountId = query.accountId as string;
```

Comment:
I feel that's outside the scope of this task but thanks for bringing it up. I created a `CurrentWorkspaceContext` react context for providing the `accountId` here => [5eb4a3d](https://github.com/hashintel/hash/pull/409/commits/5eb4a3daca1c910f2e16d65617f6824663a4dea4). Will update other instances of the code to pull the `accountId` from there in a different PR

Reply:
Is there a task for it, in which case can you link it?

And incase this isn't already your plan, can you please dedicate a single PR to a task like this, instead of addressing multiple unrelated tasks in the same PR? It allows for more concise development and reviewing, and reduces the size of PRs meaning that if a review uncovers a PR requires more work to fully address a task, it is more likely to be addressed as part of that PR - instead of being addressed in another follow-up PR.

Reply:
I just created a task for it [here](https://app.asana.com/0/1201959586244678/1202046726105145)

Reply:
Great, thank you